medallion/config.py

```diff
@@ -0,0 +1,173 @@
+import collections.abc
+import json
+import logging
+import pathlib
+import platform
+
+import appdirs
+import attr
+import environ
+import jsonmerge
+import packaging.version
+
+from . import backends
+from . import version as _medallion_version
+
+# Find a config directory for the closest compatible version including the
+# current one, or fall back to the current version's one if none are extant
+_curr_version = packaging.version.Version(_medallion_version.__version__)
+_compat_versions = {_curr_version.major, 3, 2}
```

Comment:
I included `_curr_version.major` as well as the explicit value `3` to ensure that v4 would try the v3 path unless some action is taken to avoid that (e.g. if a config schema change happens). I'm not sure what the most sensible dev flow for making a change like that would be, probably something like bumping the `medallion.version.__version__ = "4.0.0.dev0"`

Reply:
Would you rather do it this way than not including the `packaging` library and making `_compat_versions` equal to `{_medallion_version, "3", "2"}`? One less package and a little cleaner with the latter.

Reply:
Was that supposed to be `{_medallion_version.__version__, ...}`? That'd cause us to hunt for `/etc/xdg/medallion/X.Y.Z/` rather than just `.../X/` (being the current major version which will not be `3` in future).

The use of `packaging.version.Version` is to get canonical PEP440 version parsing rather than having to rely on something like `__version__.split(".", 1)[0]` which IMO is less semantically obvious.

Happy to go either way on this.

Reply:
I think you might've been right on the first try. Go with that for now and we can always change in the future if it becomes a dependency issue.